packages/jest-worker/src/Farm.ts

```diff
@@ -78,80 +81,82 @@ export default class Farm {
     });
   }
 
-  private _getNextJob(workerId: number): QueueChildMessage | null {
+  private _getNextTask(workerId: number): QueueChildMessage | null {
     let queueHead = this._queue[workerId];
 
-    while (queueHead && queueHead.request[1]) {
+    while (queueHead && queueHead.task.request[1]) {
       queueHead = queueHead.next || null;
     }
 
     this._queue[workerId] = queueHead;
 
-    return queueHead;
+    return queueHead && queueHead.task;
   }
 
   private _process(workerId: number): Farm {
-    if (this.isLocked(workerId)) {
+    if (this._isLocked(workerId)) {
       return this;
     }
 
-    const job = this._getNextJob(workerId);
+    const task = this._getNextTask(workerId);
 
-    if (!job) {
+    if (!task) {
       return this;
     }
 
     const onEnd = (error: Error | null, result: unknown) => {
-      job.onEnd(error, result);
-      this.unlock(workerId);
+      task.onEnd(error, result);
+
+      this._unlock(workerId);
       this._process(workerId);
     };
 
-    this.lock(workerId);
+    task.request[1] = true;
 
-    this._callback(workerId, job.request, job.onStart, onEnd);
-
-    job.request[1] = true;
+    this._lock(workerId);
+    this._callback(workerId, task.request, task.onStart, onEnd);
 
     return this;
   }
 
   private _enqueue(task: QueueChildMessage, workerId: number): Farm {
+    const item = {next: null, task};
+
     if (task.request[1]) {
       return this;
     }
 
     if (this._queue[workerId]) {
-      this._last[workerId].next = task;
+      this._last[workerId].next = item;
     } else {
-      this._queue[workerId] = task;
+      this._queue[workerId] = item;
     }
 
-    this._last[workerId] = task;
+    this._last[workerId] = item;
     this._process(workerId);
 
     return this;
   }
 
   private _push(task: QueueChildMessage): Farm {
     for (let i = 0; i < this._numOfWorkers; i++) {
-      const workerIdx = (this._offset + i) % this._numOfWorkers;
-      this._enqueue(task, workerIdx);
+      this._enqueue(task, (this._offset + i) % this._numOfWorkers);
     }
+
     this._offset++;
 
     return this;
   }
 
-  lock(workerId: number): void {
+  private _lock(workerId: number): void {
```

Comment:
technically breaking, but I guess these shouldn't be called anyways by consumers?

Reply:
Yeah, playing with worker locks is a recipe for disaster :D